airflow/opt/providers/etna/etna/etls/metis.py

```diff
@@ -580,7 +575,14 @@ def list_match_folders(matches):
             project_name = matches[0].project_name
             bucket_name = matches[0].bucket_name
             with self.hook.metis() as metis:
-                files_by_parent_id = { k: list(v) for k, v in itertools.groupby(metis.tail(project_name, bucket_name, 'files', folder_id=[m.folder_id for m in matches])[0], key=lambda file: file.folder_id) }
+                files_by_parent_id = {}
```

Comment:
Did not just add a `metis.tail().sort()` in case the payload is large, since sorting could be expensive and not strictly necessary.

Reply:
To be honest, in place sorting is not that expensive, it's not really clear that it would be more expensive than what you ended up doing.  (sort, other than the comparison part, is implemented natively in C, so it runs faster than the python loop).  That said, it's really not a huge deal either way.  This may just be safer implementation.

Reply:
Ah, okay, thanks, that's good to know! I wasn't sure if that would be an issue, so used the explicit / wordy way :smile: